### PR TITLE
Completed ISM specs and tests.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Completed ISM specs and tests.

### Issues Resolved

Closes https://github.com/opensearch-project/opensearch-api-specification/issues/224.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/namespaces/ism.yaml [hunk 1]
```diff
@@ -4,18 +4,56 @@ info:
...
+        - $ref: '#/components/parameters/ism.put_policies::query.policyID'
+      responses:
+        '200':
+          $ref: '#/components/responses/ism.put_policies@200'
   /_plugins/_ism/policies/{policy_id}:
     get:
       operationId: ism.get_policy.0
       x-operation-group: ism.get_policy
-      description: Gets the policy.
+      description: Creates or updates a policy.
```
Comment (nhtruong) [inline spec/namespaces/ism.yaml:36]: the group says `ism.get_policy` but the description is Create or Update?

File: spec/namespaces/ism.yaml [hunk 2]
```diff
@@ -4,18 +4,56 @@ info:
...
-        url: https://opensearch.org/docs/latest/im-plugin/ism/api/#get-policy
+        url: https://opensearch.org/docs/latest/im-plugin/ism/api/#put-policy
       parameters:
         - $ref: '#/components/parameters/ism.get_policy::path.policy_id'
       responses:
         '200':
           $ref: '#/components/responses/ism.get_policy@200'
+    head:
+      operationId: ism.exists_policy.0
+      x-operation-group: ism.exists_policy
```
Comment (nhtruong) [inline spec/namespaces/ism.yaml:46]: maybe `policy_exists`?

File: spec/namespaces/ism.yaml [hunk 3]
```diff
@@ -58,6 +110,18 @@ paths:
       responses:
         '200':
           $ref: '#/components/responses/ism.add_policy@200'
+  /_plugins/_ism/remove:
+    post:
+      operationId: ism.remove.0
+      x-operation-group: ism.remove
```
Comment (nhtruong) [inline spec/namespaces/ism.yaml:116]: This should be in the same group as `ism.remove_policy`

File: spec/namespaces/ism.yaml [hunk 4]
```diff
@@ -70,6 +134,20 @@ paths:
       responses:
         '200':
           $ref: '#/components/responses/ism.remove_policy@200'
+  /_plugins/_ism/change_policy:
+    post:
+      operationId: ism.change.0
+      x-operation-group: ism.change
```
Comment (nhtruong) [inline spec/namespaces/ism.yaml:140]: This should be in the same group as `ism_change_policy`

File: spec/namespaces/ism.yaml [hunk 5]
```diff
@@ -84,36 +162,160 @@ paths:
...
+        $ref: '#/components/requestBodies/ism.explain'
+      responses:
+        '200':
+          $ref: '#/components/responses/ism.explain@200'
   /_plugins/_ism/explain/{index}:
     get:
-      operationId: ism.explain_index.0
-      x-operation-group: ism.explain_index
+      operationId: ism.explain_policy.0
+      x-operation-group: ism.explain_policy
```
Comment (nhtruong) [inline spec/namespaces/ism.yaml:191]: Renaming groups will result in breaking changes in the generated clients. Are you sure you wanna do this?

Comment (dblock) [reply spec/namespaces/ism.yaml:191]: Yes, it's all new code. I added ISM support that has yet to be merged into any client AFAIK.

File: spec/namespaces/ism.yaml (hunk 4 above)
Comment (dblock) [reply spec/namespaces/ism.yaml:140]: But they have different parameters, one specifies the index in the query, the other in the path. So if you try to make them the same group you get this:

```
{
  file: 'namespaces/ism.yaml',
  location: 'Operation Group: ism.change_policy',
  message: "2 'ism.change_policy' operations must have an identical set of query parameters."
}
```

File: spec/namespaces/ism.yaml (hunk 3 above)
Comment (dblock) [reply spec/namespaces/ism.yaml:116]: Same as for `change_policy` they have different parameters, one specifies the index in the query, the other in the path.

File: spec/namespaces/ism.yaml [hunk 6]
```diff
@@ -44,6 +82,20 @@ paths:
       responses:
         '200':
           $ref: '#/components/responses/ism.delete_policy@200'
+  /_plugins/_ism/add:
+    post:
+      operationId: ism.add.0
+      x-operation-group: ism.add
```
[the remark this answers is not in the document]
Comment (dblock) [reply spec/namespaces/ism.yaml:88]: Same as for `change_policy` they have different parameters, one specifies the index in the query, the other in the path.

File: spec/namespaces/ism.yaml (hunk 2 above)
Comment (dblock) [reply spec/namespaces/ism.yaml:46]: We have `exists_component_template` and similar, but none of `_exists`.

File: spec/namespaces/ism.yaml (hunk 3 above)
Comment (nhtruong) [reply spec/namespaces/ism.yaml:116]: That's correct. But as far as the clients users are concerned, which is what the operation group is for, where the `index` param is put in the URL request is irrelevant. We have many API actions, like search, that also allows the index to be specified as either a path param or a query param, the generated clients will favor use it as a path param regardless.

Comment (dblock) [reply spec/namespaces/ism.yaml]: Ok, I had to add `query` to the one that doesn't really take it, but it won't matter from the client POV I think.

File: spec/namespaces/ism.yaml (hunk 4 above)
Comment (dblock) [reply spec/namespaces/ism.yaml]: Updated.

**Pull request merged**